--- antlr-3.0.1/runtime/C/dist/libantlr3c-3.0.1/src/antlr3inputstream.c	2007-10-16 16:18:59.000000000 +0200
+++ antlr-3.0.1/runtime/C/dist/libantlr3c-3.0.1/src/antlr3inputstream.c	2007-10-16 16:19:30.000000000 +0200
@@ -180,7 +180,7 @@
 
     input->nextChar		= input->data;	/* Input at first character */
     input->line			= 1;		/* starts at line 1	    */
-    input->charPositionInLine	= -1;
+    input->charPositionInLine	= 0;
     input->currentLine		= input->data;
     input->markDepth		= 0;		/* Reset markers	    */
     
